packages/blade/src/components/Link/Link/Link.tsx

```diff
@@ -0,0 +1,87 @@
+import type { ReactElement, SyntheticEvent } from 'react';
+import type { IconComponent } from '../../Icons';
+import { BaseLink } from '../BaseLink';
+
+type LinkCommonProps = {
+  variant?: 'anchor' | 'button';
+  icon?: IconComponent;
+  iconPosition?: 'left' | 'right';
+  isDisabled?: boolean;
+  onClick?: (event: SyntheticEvent) => void;
+  href?: string;
+  target?: string;
+  accessibilityLabel?: string;
+};
+
+/*
+  Mandatory children prop when icon is not provided
+*/
+type LinkWithoutIconProps = LinkCommonProps & {
+  icon?: undefined;
+  children: string;
+};
+
+/*
+  Optional children prop when icon is provided
+*/
+type LinkWithIconProps = LinkCommonProps & {
+  icon: IconComponent;
+  children?: string;
+};
+
+/*
+  Link Props with or without an icon
+*/
+type LinkPropsWithOrWithoutIcon = LinkWithIconProps | LinkWithoutIconProps;
+
+/*
+  Link Props when variant is anchor
+*/
+type LinkAnchorVariantProps = LinkPropsWithOrWithoutIcon & {
+  variant?: 'anchor';
+  href?: string;
+  target?: string;
+  rel?: string;
+  isDisabled?: undefined;
+};
+
+/*
+  Link Props when variant is button
+*/
+type LinkButtonVariantProps = LinkPropsWithOrWithoutIcon & {
+  variant?: 'button';
+  isDisabled?: boolean;
+  href?: undefined;
```

Comment:
why do we need `href` here when it is already button and other related props?

Reply:
For TS. Same like we did for Button. TS expects the prop to exist even if its undefined